### PR TITLE
Add support for running the dashboard as ingress

### PR DESCRIPTION
**Pull request by davhdavh**
### What does this PR do?

Add support for running the dashboard as ingress
Add documentation for why this may be desired
Add documentation for how to get external ip and loadbalancer ip working in own vm/hardware

### Motivation

Add some extra documentation for how to get started with traefik on a bare k8s cluster.
Easier support for ingress access to the dashboard
Implicit documentation of how to use Ingress with attributes (something the traefik docs are quite bad at showing)

### More

- [x] Yes, I updated the tests accordingly

### CODE REVIEW

**Review by mloiseleur — CHANGES_REQUESTED** [review]
Thanks for this other PR @davhdavh 
Interesting. I have two main comments:
1. We have moved all example to EXAMPLES.md, wdyt about add your example in this file ?
2. `api.insecure` is not linked to port. So this should be put elsewhere.

**Comment by davhdavh** [comment]
> api.insecure is not linked to port. So this should be put elsewhere.

https://doc.traefik.io/traefik/v3.0/operations/api/#insecure
Well, port and entryPoint is used intermixed in the helm config, but this binds specifically to the `traefik` entryPoint, and that is configured in ports.traefik. So yes I think it very much belongs here

**Comment by davhdavh** [comment]
> We have moved all example to EXAMPLES.md, wdyt about add your example in this file ?

Primarily because the main place to read the documentation does not have the examples accessible:
https://artifacthub.io/packages/helm/traefik/traefik

File: traefik/values.yaml [hunk 1]
```diff
@@ -541,6 +541,71 @@ envFrom: []
 ports:
   traefik:
     port: 9000
+    # By default do not host the dashboard on this port
+    # instead, use ingressRoute.dashboard.enabled (this uses the dashboard@internal service, instead of a seperate port)
+    # but you might need it as an ingress rather than ingressRoute, and this would allow you to do so
```
Comment (darkweaver87) [inline traefik/values.yaml:546]: Well the ingress method works but is a bit hacky.

File: traefik/values.yaml [hunk 2]
```diff
@@ -541,6 +541,71 @@ envFrom: []
...
+    # ---
+    # apiVersion: traefik.io/v1alpha1
+    # kind: Middleware
+    # metadata:
+    #   name: traefik-dashboard-auth
+    #   namespace: sameAsTraefik
+    # spec:
+    #   basicAuth:
+    #     secret: dashboard-auth          
+    # The creation of the secret resource "dashboard-auth" is left to the reader (it needs username and password fields)
```
Comment (darkweaver87) [inline traefik/values.yaml:606]:
```suggestion
    # The creation of the secret resource "dashboard-auth" is left to the reader (it needs username and password fields). You may prefer better authentication method like OIDC.
```

**Comment by darkweaver87** [comment]
> > api.insecure is not linked to port. So this should be put elsewhere.
> 
> https://doc.traefik.io/traefik/v3.0/operations/api/#insecure Well, port and entryPoint is used intermixed in the helm config, but this binds specifically to the `traefik` entryPoint, and that is configured in ports.traefik. So yes I think it very much belongs here

Well yes and no ;-) 

I would tend to agree with @mloiseleur. `api.insecure` is not in `entrypoints` section of traefik configuration and `ports` tries to stick as much as possible at it  whereas `api.insecure` deals with `routers`.

**Comment by mloiseleur** [comment]
@davhdavh Will you update this PR or should we close it ?

**Pull request closed without merging**